File: convertedToIOS5renamed/ch19p553pageController/p476pageController/Pep.m


#import "Pep.h"

@implementation Pep
@synthesize name, pic, boy;

// given a Pep boy's name, we display his name and picture

- (id) initWithPepBoy: (NSString*) inputboy nib: (NSString*) nib bundle: (NSBundle*) bundle {
    self = [self initWithNibName:nib bundle:bundle];
    if (self) {
        self.boy = inputboy;
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    self.name.text = self.boy;
    self.pic.image = [UIImage imageNamed: 
                [NSString stringWithFormat: @"%@.jpg", [self.boy lowercaseString]]];
}

- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation
{
    return (interfaceOrientation == UIInterfaceOrientationLandscapeRight);
}

@end

File: iOS8bookExamples/bk2ch06p353pageControllerRestoration3/ch19p626pageController/Pep.swift


import UIKit

class Pep: UIViewController {
    
    var boy : String // * now assignable, we will assign on decode
    @IBOutlet var name : UILabel!
    @IBOutlet var pic : UIImageView!
    
    // we add "required" to satisfy the compiler's worry that class's "self" might be a subclass
    required init(pepBoy boy:String) {
        self.boy = boy
        super.init(nibName: "Pep", bundle: nil)
        self.restorationIdentifier = "pep" // *
        // self.restorationClass = self.dynamicType // * no restoration class, let app delegate point
    }
    
    required init(coder: NSCoder) {
        fatalError("NSCoding not supported")
    }
    
    override func encodeRestorableStateWithCoder(coder: NSCoder) {
        // super.encodeRestorableStateWithCoder(coder)
        println("pep about to encode boy \(self.boy)")
        coder.encodeObject(self.boy, forKey:"boy")
    }
    
    // and now we simply decode directly
    
    override func decodeRestorableStateWithCoder(coder: NSCoder) {
        // super.decodeRestorableStateWithCoder(coder)
        let boy : AnyObject? = coder.decodeObjectForKey("boy")
        println("pep about to decode boy \(boy)")
        if let boy = boy as? String {
            self.boy = boy
        }
    }
    
    // one more step; in case we just did state restoration,
    // match interface to newly assigned boy
    // oops, not DRY: this is exactly the same as viewDidLoad()
    
    override func applicationFinishedRestoringState() {
        self.name.text = self.boy
        self.pic.image = UIImage(named:"\(self.boy.lowercaseString).jpg")
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        self.name.text = self.boy
        self.pic.image = UIImage(named:"\(self.boy.lowercaseString).jpg")
    }
    
    override var description : String {
    return self.boy
    }
    
    @IBAction func tap (sender: UIGestureRecognizer?) {
        NSNotificationCenter.defaultCenter().postNotificationName("tap", object: sender)
    }
}

// and no UIViewControllerRestoration

